src/mscorlib/src/System/String.cs

```diff
@@ -1208,6 +1272,42 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
             }
             return foundCount;
         }
+
+        private struct SplitSeparatorArray
+        {
+            private readonly bool _single;
```

Comment:
`string.Empty` is a valid value as well.

For example, consider the calls to the following existing `Split` method:

```c#
"a,b".Split(new string[] { null }, StringSplitOptions.None);
"a,b".Split(new string[] { string.Empty }, StringSplitOptions.None);
```

Both calls are valid. If we want the new overloads to behave the same as the existing methods, just as if a single separator was passed, then the following needs to behave the same as above:

```c#
"a,b".Split((string)null, StringSplitOptions.None);
"a,b".Split(string.Empty, StringSplitOptions.None);
```

When `new SplitSeparatorArray((string)null)` or `new SplitSeparatorArray(string.Empty)` is used, it needs to behave the same as `new SplitSeparatorArray(new string[] { null })` or `new SplitSeparatorArray(new string[] { string.Empty })`.

Passing in a null or empty `string[]` is also valid, so you can't switch on that either.